packages/venia-concept/src/ServiceWorker/registerRoutes.js

```diff
@@ -61,10 +67,19 @@ export default function() {
      * Route for all JS files and bundles. This route uses CacheFirst
      * strategy because if the file contents change, the file name will
      * change. There is no point in using StaleWhileRevalidate for JS files.
+     *
+     * After 24 hours, the files will be purged from cache.
      */
     workbox.routing.registerRoute(
         new RegExp(/\.js$/),
-        new workbox.strategies.CacheFirst()
+        new workbox.strategies.CacheFirst({
+            cacheName: RUNTIME_CACHE_NAME,
+            plugins: [
+                new workbox.expiration.Plugin({
+                    maxAgeSeconds: ONE_DAY
```

Comment:
Expire all JS files after 24 hours. If nothing has changed, those files will be fetched only once a day till a new version is available. It is just to make sure we do not have stale JS files lingering around in cache bloating it. It is totally configurable to someone's taste.

Reply:
I think "stale" JS files is probably the wrong term. As a PWA we want to be accessible offline, which means we should ideally cache files indefinitely (or until the next release).

Is there some other solution to this bug that could involve invalidating the index file instead?

Reply:
You are absolutely right, the whole point of a PWA is to be accessible every time.

The timeout I have added is only to delete old files. We can simply omit that or change it to a longer period. The code should still work as expected.

Also, I have changed it to 1 week now. I am open to removing it all together.

Reply:
@sirugh check this out https://github.com/magento/pwa-studio/pull/2376/commits/0111b07b91caff0d0ff6b66cb2cb0cded8aae06e#diff-62affb4ad3e6b6a3d145bfdd51b6178fL284

My first approach was to delete the files when a new HTML file is discovered, that way we will never have any offline issues, but workbox has some weird race condition issues due to which it used to work intermittently. Hence I added the `maxAgeSeconds` variable and letting workbox delete the files for me.